src/main/java/NumberGenerator.java

```diff
@@ -0,0 +1,32 @@
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Set;
+
+public class NumberGenerator {
+    private final Random random;
+
+    NumberGenerator() {
+        this.random = new Random();
+    }
+
+    // [lo, hi] 범위의 난수 하나 생성
+    public int generateOneNumber(int lo, int hi) {
+        return random.nextInt(hi - lo + 1) + lo;
+    }
+
+    // [lo, hi] 범위의 난수 length개 생성
+    public int[] generateNumbers(int length, int lo, int hi) {
+        int[] result = new int[length];
+        Set<Integer> existNumberSet = new HashSet<>();
+        int ix = 0;
+        while (ix < length) {
+            int number = generateOneNumber(lo, hi);
+            if (!existNumberSet.contains(number)) {
+                result[ix] = number;
+                existNumberSet.add(number);
+                ix++;
+            }
+        }
+        return result;
+    }
```

Comment:
이렇게 줄일 수도 있어보이는데 어떤가요?
contains로 직접 체크해본 이유가 있는지 궁금합니다.

(원한다면 stream 도 사용할 수 있어보이지만, 효율적인지는 모르겠네요.)

```suggestion
    public int[] generateNumbers(int length, int lo, int hi) {
        int[] result = new int[length];
        Set<Integer> existNumberSet = new HashSet<>();
        while (existNumberSet.size() < length) {
            int number = generateOneNumber(lo, hi);
            existNumberSet.add(number);
        }
        return result;
    }
```

Reply:
existNumberSet에 존재하지 않을 경우에만 result에 추가하는 로직이 필요해서 기존처럼 구성했어요.
제안한 방식대로 구현하고 list로 바꾸는 방법도 생각해봤는데 Hash 특성상 일정한 순서대로 정렬될 가능성이 있어서 배열에 따로 넣는 과정을 선택했어요!

Reply:
순서를 지켜주는 `LinkedHashSet`이 있긴 한데... 자바의 영역이니 언급만 하겠습니다 ㅋㅋ